aten/src/ATen/functorch/ADInterpreters.cpp

```diff
@@ -136,12 +155,21 @@ static void autogradBasedTransformSendToNext(
   if (getDynamicLayerStack().size() == 0) {
     sanityCheckStack(op, stack);
   }
-  op.callBoxed(stack);
 
   // Step 4, 5, 6
   auto ret_size = op.schema().returns().size();
+
+  op.callBoxed(stack);
+
+  // lift_fresh: it's must be freshly allocated and should be wrapped. User shouldn't have access to input version
+  // alias: this is needed for the CompositeImplicit instance norm (running_mean/var get set to be a wrapped value)
+  //        It's not a user facing function, but is more prone to possible errors
+  int64_t outputs_aliasing_immutable = 0;
+  if (!grad_special_case || immutable_inputs == 0) {
+    outputs_aliasing_immutable = findAliasedOutputs(op.schema(), immutable_inputs);
+  }
```

Comment:
nit: There's no reason to split the `immutable_inputs` computation code and the `outputs_aliasing_immutable` computation code, right? Putting them together would group the logic in a way that we could refactor it into a helper function that, given the inputs and the schema, computes which outputs should be immutable.